spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/health/PubSubHealthIndicatorAutoConfiguration.java

```diff
@@ -46,6 +47,7 @@
 	@Bean
 	@ConditionalOnMissingBean
 	public PubSubHealthIndicator pubSubHealthIndicator(PubSubTemplate pubSubTemplate) {
+		Assert.notNull(pubSubTemplate, "PubSubTemplate can't be null");
```

Comment:
This is redundant. You have a check in the `PubSubHealthIndicator`.
Also this is always not going to fail: an injection without `@Autowired(required = false)` is always going to fail for non-existing bean.
So, you are overengineering here 😄

Reply:
True @artembilan thanks again for your thoughts, I guess I did over engineer it, but it would catch non bean usages too? If by any chance someone is instantiating the class outside of the container. I know its a far stretch but my point is, are you ok with the extra check so we can merge?

Reply:
I'm not sure what you mean, but what I see here is this:
```
public PubSubHealthIndicator pubSubHealthIndicator(PubSubTemplate pubSubTemplate) {
     return new PubSubHealthIndicator(pubSubTemplate);
}
```

Even if someone will call this method outside of the container, we still have an NPE check in the `PubSubHealthIndicator` ctor.

So, I still see how this `Assert` is redundant.